src/main/java/software/amazon/awssdk/crt/http/HttpStream.java

```diff
@@ -55,6 +63,57 @@ public void incrementWindow(int windowSize) {
         }
     }
 
+    /**
+     * Use only for Http 1.1 Chunked Encoding. At some later point we may adapt this interface for H2, but not yet. If
+     * using chunked encoding, remember, you'll need to send an empty chunk to signal the end of your stream. You must call
```

Comment:
I think this distinction will push unnecessary complexity into customer code. If you use a `Content-Length`, can you publish an empty chunk at the end of the stream?